src/dispatch/plugins/dispatch_slack/modals.py

```diff
@@ -304,3 +320,91 @@ def create_update_participant_modal(incident_id: int, command: dict, db_session=
     )
+
+
+def build_update_notifications_group_blocks(incident: Incident):
+    """Builds all blocks required to update the membership of the notifications group."""
+    modal_template = {
+        "type": "modal",
+        "title": {"type": "plain_text", "text": "Update Group Membership"},
+        "blocks": [
+            {
+                "type": "context",
+                "elements": [
+                    {
+                        "type": "plain_text",
+                        "text": "Use this form to update the membership of the notifications group.",
+                    }
+                ],
+            },
+        ],
+        "close": {"type": "plain_text", "text": "Cancel"},
+        "submit": {"type": "plain_text", "text": "Update"},
+        "callback_id": UpdateNotificationsGroupCallbacks.submit_form,
+        "private_metadata": json.dumps({"incident_id": str(incident.id)}),
+    }
+
+    group_plugin = plugins.get(INCIDENT_PLUGIN_GROUP_SLUG)
+    members = group_plugin.list(incident.notifications_group.email)
+
+    members_block = {
+        "type": "input",
+        "block_id": UpdateNotificationsGroupBlockFields.update_members,
+        "label": {"type": "plain_text", "text": "Members"},
+        "element": {
+            "type": "plain_text_input",
+            "action_id": UpdateNotificationsGroupBlockFields.update_members,
+            "multiline": True,
+            "initial_value": (", ").join(members),
+        },
+    }
+    modal_template["blocks"].append(members_block)
+
+    modal_template["blocks"].append(
+        {
+            "type": "context",
+            "elements": [{"type": "plain_text", "text": "Separate email addresses with commas."}],
+        },
+    )
+
+    return modal_template
+
+
+@background_task
+def create_update_notifications_group_modal(incident_id: int, command: dict, db_session=None):
+    """Creates a modal for editing members of the notifications group."""
+    trigger_id = command["trigger_id"]
+
+    incident = incident_service.get(db_session=db_session, incident_id=incident_id)
+
+    modal_create_template = build_update_notifications_group_blocks(incident=incident)
+
+    dispatch_slack_service.open_modal_with_user(
+        client=slack_client, trigger_id=trigger_id, modal=modal_create_template
+    )
+
+
+@background_task
+def update_notifications_group_from_submitted_form(action: dict, db_session=None):
+    """Updates notifications group based on submitted form data."""
+    submitted_form = action.get("view")
+    parsed_form_data = parse_submitted_form(submitted_form)
+
+    current_members = (
+        submitted_form["blocks"][1]["element"]["initial_value"].replace(" ", "").split(",")
+    )
+    updated_members = (
+        parsed_form_data.get(UpdateNotificationsGroupBlockFields.update_members)
+        .replace(" ", "")
+        .split(",")
+    )
+
+    members_added = numpy.setdiff1d(updated_members, current_members)
```

Comment:
This feels a bit weird, wouldn't it be better to use something out of the standard library? https://docs.python.org/3.7/library/stdtypes.html#frozenset.difference